controllers/posts.js

```diff
@@ -30,7 +33,80 @@ export default class PostController {
         post: dataValues,
       });
     } catch (error) {
-      return res.status(500).send({ message: error.stack, status: 500 });
+      return res.status(500).send({ message: error, status: 500 });
+    }
+  }
+
+  static async editPost(req, res) {
+    try {
+      const { postId } = req.params;
+      const userId = req.user.id;
+      const {
+        title, content, views, mediaPath,
+      } = req.body;
+
+      const postResponse = await Post.findOne({
+        where: {
+          id: postId,
+          userId,
+        },
+      });
+
+      if (!postResponse) {
+        return res.status(404).send({ message: 'The post does not exist', status: 404 });
+      }
+      const { dataValues } = await postResponse.update({
+        title, content, views, mediaPath, updatedAt: moment().format(),
+      });
+
+      return res.status(200).send({
+        message: 'The post was updated successfully',
+        status: 200,
+        post: dataValues,
+      });
+    } catch (error) {
+      return res.status(500).send({ message: error, status: 500 });
+    }
+  }
+
+  static async getOnePost(req, res) {
+    try {
+      const { postId } = req.params;
+
+      const postResponse = await Post.findOne({
+        where: {
+          id: postId,
+        },
+      });
+
+      if (!postResponse) {
+        return res.status(404).send({ message: 'The post does not exist', status: 404 });
+      }
+
+      return res.status(200).send({
+        message: 'A post fetched successfully',
+        status: 200,
+        post: postResponse,
+      });
+    } catch (error) {
+      return res.status(500).send({ message: error, status: 500 });
+    }
+  }
+
+  static async getAllPost(req, res) {
+    try {
+      const postResponse = await Post.findAll();
```

Comment:
you can include author details on each post.

Reply:
@abayo-luc  Done!